pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -430,6 +441,38 @@ func (h *Health) checkControlPlane(
 	return &c, nil
 }
 
+// CheckIfDependencyWatchdogProberScaledDownControllers checks if controllers have been scaled down by dependency-watchdog-prober.
+func CheckIfDependencyWatchdogProberScaledDownControllers(ctx context.Context, seedClient client.Client, shootNamespace string) ([]string, error) {
+	var scaledDownDeploymentNames []string
+
+	proberConfig, err := kubeapiserver.NewDependencyWatchdogProberConfiguration()
+	if err != nil {
+		return nil, fmt.Errorf("failed getting dependency-watchdog-prober config: %w", err)
+	}
+
+	for _, dependentResourceInfo := range proberConfig {
+		if dependentResourceInfo.Ref == nil || dependentResourceInfo.Ref.Kind != "Deployment" {
+			continue
+		}
+
+		deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: dependentResourceInfo.Ref.Name, Namespace: shootNamespace}}
+		if err := seedClient.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
+			if apierrors.IsNotFound(err) {
```

Comment:
```suggestion
			if apierrors.IsNotFound(err) && dependentResourceInfo.Optional == true {
```

We should continue only if it is optional, right? If `kcm` or `mcm` deployment is not present, shouldn't this check return an error?